src/main/java/it/pagopa/pn/deliverypush/service/mapper/NotificationMapper.java

```diff
@@ -0,0 +1,259 @@
+                                                            .sha256(payment.getF24flatRate().getDigests().getSha256())
+                                                            .build()
+                                            )
+                                            .ref(
+                                                    NotificationDocumentInt.Ref.builder()
+                                                            .key(payment.getF24flatRate().getRef().getKey())
+                                                            .versionToken(payment.getF24flatRate().getRef().getVersionToken())
+                                                            .build()
+                                            )
+                                            .build()
+                            );
+                }
+                notificationRecIntBuilder.payment(paymentInfoBuilder.build());
+            }
+            list.add(notificationRecIntBuilder.build());
+        }
+        return list;
+    }
+
+
+
+    public static SentNotification internalToExternal(NotificationInt notification) {
+        SentNotification sentNotification = new SentNotification();
+
+        sentNotification.setIun(notification.getIun());
+        sentNotification.setPaProtocolNumber(notification.getPaNotificationId());
+        sentNotification.setSentAt(notification.getSentAt());
+
+        List<NotificationRecipient> recipients = notification.getRecipients().stream()
+                .map(NotificationMapper::getNotificationRecipient).collect(Collectors.toList());
+
+        sentNotification.setRecipients(recipients);
+
+        List<NotificationDocument> documents = notification.getDocuments().stream().map(
+                NotificationMapper::getNotificationDocument).collect(Collectors.toList());
+
+        sentNotification.setDocuments(documents);
+
+        if(notification.getPhysicalCommunicationType() != null){
+            sentNotification.setPhysicalCommunicationType(SentNotification.PhysicalCommunicationTypeEnum.valueOf(notification.getPhysicalCommunicationType().name()));
+        }
+
+        return sentNotification;
+    }
+
+    @NotNull
+    private static NotificationDocument getNotificationDocument(NotificationDocumentInt documentInt) {
+        NotificationAttachmentDigests digests = new NotificationAttachmentDigests();
+        digests.setSha256(documentInt.getDigests().getSha256());
+
+        NotificationAttachmentBodyRef ref = new NotificationAttachmentBodyRef();
+        ref.setKey(documentInt.getRef().getKey());
+        ref.setVersionToken(documentInt.getRef().getVersionToken());
+
+        NotificationDocument document = new NotificationDocument();
+        document.setDigests(digests);
+        document.setRef(ref);
+        return document;
+    }
+
+    @NotNull
+    private static NotificationRecipient getNotificationRecipient(NotificationRecipientInt recipient) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This method contains 53 lines of code, not including blank lines or lines with only comments, Java punctuation characters, identifiers, or literals. By comparison, 98% of the methods in the CodeGuru reference dataset contain fewer lines of code. Large methods might be difficult to read and have logic that is hard to understand and test. We recommend that you simplify this method or break it into multiple methods.